v0/destinations/google_adwords_remarketing_lists/util.js

```diff
@@ -0,0 +1,179 @@
+const { httpSend } = require("../../../adapters/network");
+const { isHttpStatusSuccess } = require("../../util/index");
+const { TRANSFORMER_METRIC } = require("../../util/constant");
+const ErrorBuilder = require("../../util/error");
+const {
+  DISABLE_DEST,
+  REFRESH_TOKEN
+} = require("../../../adapters/networkhandler/authConstants");
+
+const {
+  processAxiosResponse
+} = require("../../../adapters/utils/networkUtils");
+
+/**
+ * This function helps to create a offlineUserDataJobs
+ * @param endpoint
+ * @param customerId
+ * @param listId
+ * @param headers
+ * @param method
+ */
+
+const createJob = async (endpoint, customerId, listId, headers, method) => {
+  const jobCreatingUrl = `${endpoint}:create`;
+  const jobCreatingRequest = {
+    url: jobCreatingUrl,
+    data: {
+      job: {
+        type: "CUSTOMER_MATCH_USER_LIST",
+        customerMatchUserListMetadata: {
+          userList: `customers/${customerId}/userLists/${listId}`
+        }
+      }
+    },
+    headers,
+    method
+  };
+  const response = await httpSend(jobCreatingRequest);
+  return response;
+};
+/**
+ * This function helps to put user details in a offlineUserDataJobs
+ * @param endpoint
+ * @param headers
+ * @param method
+ * @param jobId
+ * @param body
+ */
+
+const addUserToJob = async (endpoint, headers, method, jobId, body) => {
+  const jobAddingUrl = `${endpoint}/${jobId}:addOperations`;
+  const secondRequest = {
+    url: jobAddingUrl,
+    data: body.JSON,
+    headers,
+    method
+  };
+  const response = await httpSend(secondRequest);
+  return response;
+};
+
+/**
+ * This function helps to run a offlineUserDataJobs
+ * @param endpoint
+ * @param headers
+ * @param method
+ * @param jobId
+ */
+const runTheJob = async (endpoint, headers, method, jobId) => {
+  const jobRunningUrl = `${endpoint}/${jobId}:run`;
+  const thirdRequest = {
+    url: jobRunningUrl,
+
+    headers,
+    method
+  };
+  const response = await httpSend(thirdRequest);
+  return response;
+};
+
+/**
+ * This function is responsible for making the three steps required for uploding
+ * data to customer list.
+ * @param {*} request
+ * @returns
+ */
+const gaAudienceProxyRequest = async request => {
+  const { body, method, params, endpoint } = request;
+  const { headers } = request;
+  const { customerId, listId } = params;
+
+  // step1: offlineUserDataJobs creation
+
+  const firstResponse = await createJob(
+    endpoint,
+    customerId,
+    listId,
+    headers,
+    method
+  );
+  if (!isHttpStatusSuccess(firstResponse.response.status)) return firstResponse;
+
+  // step2: putting users into the job
+  const jobId = firstResponse.response.data.resourceName.split("/")[3];
```

Comment:
We should add some checks for 200 but in the data, there could be some error messages.

Reply:
Done